persistence/selecting.md

```diff
@@ -0,0 +1,103 @@
+## Persister decision chart
+
+As guidance, this decision chart can be used to select an appropriate persister. It is by no means an end to all discussions and additional considerations have been added to this document, to make the best decision possible.
+
+```mermaid
+graph TD
+start((start)) --> A{Using sagas<br /> or outbox?}
+A -->  |Yes|B{Additional<br />requirements?}
+A --> |No| I{Which transport?}
+I --> |Other|L[SQL Persistence]
+I --> |RabbitMQ/ASB|J[No persistence]
+I --> |ASQ|K[ASP]
+B --> |ServiceFabric|C[ServiceFabric]
+B --> |RavenDb|E[RavenDb]
+B --> |Azure|Q{IAAS/PAAS ?}
+B --> |No|L
+Q --> |PAAS|S[ASP]
+Q --> |IAAS|L
+```
+## Making the decision
+
+This section provides considerations that aren’t included or immediately visible in the decision chart. The selected transport might influence the decision for a persister. See the [selecting a transport guidance](/transports/selecting.md) for more information.
+
+Most users will eventually start recognizing the benefits of sagas in their business system. For this reason, it is important for any system to consider selecting a persister.
+
+If a datastore has already been selected to store business data and NServiceBus has a persister for that specific datastore, it can be beneficial to select that specific persister. If NServiceBus doesn’t have a persister for the specific data store or multiple data stores are used, any of the NServiceBus persisters can be used. The result is a hybrid solution of multiple data stores to store both business data and NServiceBus specific data.
+
+### Learning Persistence
+
+The Learning Persistence is not meant for production usage.
+
+### In-Memory
+
+The In-Memory persister is not mentioned in the decision chart, as it is only usable in very specific scenarios where the volatility of data is not an issue. 
+
+### Azure
+
+There are multiple options available when endpoints are hosted in Windows Azure. The most commonly used persisters are Azure Storage and Azure SQL, the fully managed SQL Server solution in Azure.
+
+The decision should be made whether you want to be fully Platform-as-a-Service (PaaS) enabled - and want to fully buy-in to Azure - or if you’re confident managing SQL Server yourself you may choose Infrastructure-as-a-Service (IaaS). Although there are other small considerations to be made for either option.
+
+Azure SQL
+
+- Supports local transactions
+- Supports Outbox
+- Better tooling support with SQL Management Studio
+
+Azure Storage
+
+- Cheaper
+- Automatically scales
+- A turn-key solution, meaning no maintenance
+
+### SQL Persistence
+
+This persister stores saga data as a JSON blob. This gives much better performance and maintainability than the NHibernate persister. The NHibernate persister stores data in tabular format, creating multiple tables for sagas, especially when using complex data types.
+
+Since SQL Server and PostgreSQL have great support for querying JSON data, this persister is recommended over NHibernate.
```

Comment:
Perhaps have a section for NHibernate that explicitly says "For new development, this persister is not recommended. Instead, use the SQL persister."

Similarly, might be a good idea to include MongoDB. Both this and NHibernate are listed as top-level options in the Persistence menu.

Reply:
In the PoA there is an action point to create a new issue for adding Mongo, since our own version is not released yet. I'll look into the NHibernate section.